tests/Constraint/ConstraintTest.php

```diff
@@ -158,4 +158,60 @@ public function invalidOperators()
             array('1.2.3', 'equals', 'InvalidArgumentException'),
         );
     }
-}
+
+    /**
+     * @dataProvider bounds
+     *
+     * @param string $operator
+     * @param string $normalizedVersion
+     * @param array $expectedBounds
+     */
+    public function testBounds($operator, $normalizedVersion, array $expectedBounds)
+    {
+        $constraint = new Constraint($operator, $normalizedVersion);
+
+        $this->assertSame($expectedBounds, $constraint->getBounds());
+    }
+
+    /**
+     * @return array
+     */
+    public function bounds()
+    {
+        return array(
+            'equal to 1.0.0.0' => array('==', '1.0.0.0', array(array('lower' => '1.0.0.0', 'upper' => '1.0.0.0'))),
+
+            'lower than 0.0.4.0' => array('<', '0.0.4.0', array(array('lower' => '0', 'upper' => '0.0.3.9999999'))),
```

Comment:
I think it may actually be easier to just define wether the bound is inclusive or exclusive, than to calculate the lower number.

So e.g. `< 0.0.4.0` would resolve to `['lower' => [INCLUSIVE, '0'], 'upper' => [EXCLUSIVE, '0.0.4.0']]` while `<= 0.0.4.0` would resolve to `['lower' => [INCLUSIVE, '0'], 'upper' => [INCLUSIVE, '0.0.4.0']]`, with some INCLUSIVE/EXCLUSIVE constants defined as true/false I suppose. Rather than using 0 and all nines as min/max you could also literally use INT_MIN/INT_MAX or some constant representing these, then one can simply skip checking the lower/upper bound entirely if that is the value.

Reply:
Okay, given that fact that you only want to use it for `version_compare()` directly, we might as well just provide the value and the operator.
So given we had `< 7.4`, `$constraint->getLowerBound()` would return `['>=', '0']` and `$constraint->getUpperBound()` would return `['<', '7.4']`. Then you can use the result to generate your `version_compare()` statements directly without even handling some `INCLUSIVE` or `EXCLUSIVE` constants?
The only thing that remains an issue is the `9999999...........9999999` upper bound which I would just replace by an infinity constant so one can skip those because they will always be true.
I don't think we need one for the lower boundary because a version lower than 0 just doesn't exist.
Wdyt?

Reply:
Just using the operator seems reasonable too! I agree with you on infinity/0 too!

Reply:
So it should be as simple as the latest 1aa9738 now.